ZenPacks/zenoss/ZenPackLib/lib/base/types.py

```diff
@@ -112,7 +112,7 @@ class Severity(int):
                'clear': 0}
 
     def __new__(cls, value):
-        if not value:
+        if not value and not isinstance(value, int):
```

Comment:
seems like we should actually just allow for any value to flow through to validate() and then issue a warning there saying it's invalid?  if it's not a valid int or str, then we could default it to 3 and issue the warning.  or even raise an exception

```
    def __new__(cls, value):
        return int.__new__(cls, cls.validate(value))

    @classmethod
    def validate(cls, value):
        try:
            value = int(value)
            if value < 0:
                cls.LOG.warning("Invalid severity value ({}), increasing to 0".format(value))
                value = 0
            elif value > 5:
                cls.LOG.warning("Invalid severity value ({}), reducing to 5".format(value))
                value = 5
        except (TypeError, ValueError):
            if isinstance(value, str) and value.lower() in cls._valid_text:
                value = cls.to_num.get(value.lower())
            else:
                # return 3 or raise exception here?
                sev_num_txt = [str(x) for x in cls._valid_num]
                cls.LOG.warning("Invalid severity value ({}), "\
                    "must be one of: ({}) or ({}).".format(value,
                                            ', '.join(cls._valid_text),
                                            ', '.join(sev_num_txt)))
                value = 3
        return value
```